torch/csrc/api/src/nn/modules/embedding.cpp

```diff
@@ -11,27 +11,203 @@
+  c10::optional<torch::Tensor> per_sample_weights) {
+
+    TORCH_CHECK(per_sample_weights == c10::nullopt || ((input.size(0) == per_sample_weights.size(0)) && input.size(1) == per_sample_weights.size(1)),
+      "embedding_bag: If per_sample_weights ({", per_sample_weights.size(0), ", ", per_sample_weights.size(1), "}) is not null,
+                            then it must have the same shape as the input ({", input.size(0), ", ", input.size(1), "})\n");
+    if(input.dim() == 2) {
+      TORCH_CHECK(offsets == c10::nullopt,
+        "if input is 2D, then offsets has to be null, as input is treated is a mini-batch of
+                    fixed length sequences. However, found an offsets Tensor"); //check about adding type
+        offsets = torch::arange(0, input.numel(), input.size(1),
+                                   torch::TensorOptions().dtype(torch::kLong).device(input.device()))
+        input = input.reshape(-1);
+        if(per_sample_weights != c10::nullopt) {
+          per_sample_weights = per_sample_weights.reshape(-1);
+        }
+      }
+    }
+    else if(input.dim() == 1) {
+      TORCH_CHECK(offsets != c10::nullopt, "offsets has to be a 1D Tensor but got null");
+      TORCH_CHECK(offsets.dim() == 1, "offsets has to be a 1D Tensor");
+      TORCH_CHECK(offsets.dim() == 0, "offsets[0] has to be 0, i.e., the first sequence in the mini-batch has to start from position 0.
+                However, got ", int(offsets[0]), "\n");
+      TORCH_CHECK(int(offsets[offsets.size(0)-1]) <= input.size(0), "offsets[offsets.size(0)-1] can not be greater than input's length({)",
+                input.size(0), "}), but got offsets[offsets.size(0)-1] of {", int(offsets[offsets.size(0)-1]), "}\n");
+      }
+    }
+    else{
+      TORCH_CHECK(false, "input has to be 1D or 2D Tensor,but got Tensor of dimension {", input.dims(), "}\n");
+    }
+
+    int mode_enum;
+    if(mode == "sum") {
+      mode_enum = 0;
+    }
+    else if(mode == "mean") {
+      mode_enum = 1;
+    }
+    else if(mode =="max") {
+      mode_enum = 2;
+      TORCH_CHECK(!options.scale_grad_by_freq(), "max mode does not support scaling the gradient by the frequency");
+      TORCH_CHECK(!sparse, "max mode does not support sparse weights");
+    }
+    else{
+      TORCH_CHECK(false, "mode has to be one of sum, mean or max");
+    }
+
+    if(options.max_norm() != c10::nullopt) {
+      torch::NoGradGuard no_grad;
+      torch::embedding_renorm_(weight, input, *options.max_norm(), options.norm_type());
+    }
+
+    TORCH_CHECK((per_sample_weights == c10::nullopt) || (mode == "sum"), "embedding_bag: per_sample_weights was not null
+          per_sample_weights is only supported for mode='sum' (got mode='{", mode,
+          "})Please open a feature request on GitHub.");
+    return torch::embedding_bag(weight, input, offsets, scale_grad_by_freq, mode_enum, sparse, per_sample_weights);
```

Comment:
Since `torch::embedding_bag` returns a tuple, we need:
```suggestion
    return std::get<0>(torch::embedding_bag(weight, input, offsets, scale_grad_by_freq, mode_enum, sparse, per_sample_weights));
```